runtime/compiler/optimizer/InterpreterEmulator.cpp

```diff
@@ -594,7 +598,15 @@ InterpreterEmulator::prepareToFindAndCreateCallsites(TR::Block **blocks, flags8_
 void
 InterpreterEmulator::visitInvokedynamic()
    {
-   int32_t cpIndex = next2Bytes();
+   TR_ResolvedJ9Method * owningMethod = static_cast<TR_ResolvedJ9Method*>(_methodSymbol->getResolvedMethod());
+   int32_t callSiteIndex = next2Bytes();
+#if defined(J9VM_OPT_OPENJDK_METHODHANDLE)
+   if (owningMethod->isUnresolvedCallSiteTableEntry(callSiteIndex) ||
+         comp()->compileRelocatableCode() ||
+         comp()->isOutOfProcessCompilation()) return; // do nothing if unresolved, is AOT or JITServer compilation
```

Comment:
@dmitry-ten Should this query be guarded by `J9VM_OPT_JITSERVER` macro?

Reply:
Right, it should be.